src/middlewared/middlewared/plugins/reporting/netdata_configure.py

```diff
@@ -24,17 +24,20 @@ def netdata_state_location(self):
 
     @private
     def post_dataset_mount_action(self):
-        if os.path.exists(get_netdata_state_path()):
+        netdata_state_path = get_netdata_state_path()
+        if os.path.exists(netdata_state_path):
             return
 
         cp = subprocess.run(
-            ['cp', '-a', '/var/lib/netdata', get_netdata_state_path()], check=False, capture_output=True,
+            ['cp', '-a', '/var/lib/netdata', netdata_state_path], check=False, capture_output=True,
```

Comment:
Why are we doing fork+exec'ing here? Why not use `shutil.rmtree`?

Reply:
Woops, not `shutil.rmtree`. I meant to say `shutil.copy2` :smile:

Reply:
We can use `middleware.utils.filesystem.copy.copytree` as well.